src/main/diagnostics/dockerCliSymlinks.ts

```diff
@@ -40,71 +40,85 @@ export class CheckerDockerCLISymlink implements DiagnosticsChecker {
-      if (link === rdBinPath) {
-        while (true) {
-          try {
-            link = await this.readlink(link);
-          } catch (ex) {
-            break;
-          }
-        }
-        console.debug(`docker-cli symlink: ${ this.name }: final symlink: ${ link } (app dir ${ paths.resources })`);
-        if (path.relative(paths.resources, link).startsWith('.')) {
-          state = `is a symlink to ${ replaceHome(link) }, which is not from Rancher Desktop`;
-        } else {
-          try {
-            await this.access(link, fs.constants.X_OK);
-            state = `is a symlink to ~/.rd/${ this.name }`;
-            passed = true;
-          } catch (ex) {
-            const code = (ex as any).code ?? '';
-
-            if (code === 'ENOENT') {
-              state = `is a symlink to ${ replaceHome(link) }, which does not exist`;
-            } else if (code === 'ELOOP') {
-              state = `is a symlink with a loop`;
-            } else if (code === 'EACCES') {
-              state = `is a symlink to ${ replaceHome(link) }, which is not executable`;
-            } else {
-              state = `is a symlink to ${ replaceHome(link) }, but we could not read it (${ code })`;
-            }
-          }
-        }
-      } else {
-        state = `is a symlink to ${ replaceHome(link) }`;
+      link = await this.readlink(startingPath);
+
+      console.debug(`docker-cli symlink: ${ this.name }: first-level symlink ${ startingPath }: points to: ${ link } (expect ${ rdBinPath })`);
+
+      if (link !== rdBinPath) {
+        return {
+          description: `${ description } should be a symlink to ${ replaceHome(rdBinPath) }, but points to ${ link }.`,
+          passed:      false,
+          fixes:       [], // TODO: [{ description: `ln -sf ${ replaceHome(rdBinPath) } ${ replaceHome(startingPath) }` }],
+        };
       }
-    } catch (ex) {
+    } catch (ex: any) {
       const code = (ex as any).code ?? '';
 
-      console.debug(`docker-cli symlink: ${ this.name } got exception ${ code }: ${ ex }`);
       if (code === 'ENOENT') {
-        state = `does not exist`;
+        state = 'does not exist';
       } else if (code === 'EINVAL') {
         state = `is not a symlink`;
       } else {
-        state = `cannot be read`;
+        state = 'cannot be read';
       }
-    }
-    let description = `The file ~/.docker/cli-plugins/${ this.name } ${ state }.`;
 
-    if (!passed) {
-      description += `  It should be a symlink to ~/.rd/bin/${ this.name }.`;
+      return {
+        description: `${ description } ${ state }. It should be a symlink to ${ replaceHome(rdBinPath) }.`,
+        passed:      false,
+        fixes:       [],
+      };
     }
 
-    return {
-      description,
-      passed,
-      fixes: [],
-    };
+    description = `The file ${ rdBinPath }`;
+    try {
+      link = await this.readlink(link);
```

Comment:
This assignment to itself is confusing; at least, I find `this.readlink(rdBinPath)` to be much clearer (because that's a constant value, and more specifically named).

If we do that though, it'd make sense to make `link` scoped to within the `try` for both instances instead (`const link = await this.readlink(startingPath);` and `const link = await this.readlink(rdBinPath);` — that has the advantage that in the second `catch` clause it becomes really clear that `… is a symlink to ${ replaceHome(link) }` is really referring to `rdBinPath` instead (because the assignment never happened).

Reply:
Good point. No need to have a mutable `link` given that we're no longer walking loops.